components/eamxx/src/physics/nudging/atmosphere_nudging.cpp

```diff
@@ -0,0 +1,264 @@
+#include "atmosphere_nudging.hpp"
+
+namespace scream
+{
+
+  //using namespace spa;
+// =========================================================================================
+NUDGING::NUDGING (const ekat::Comm& comm, const ekat::ParameterList& params)
+  : AtmosphereProcess(comm, params)
+{
+  m_fnames=m_params.get<std::vector<std::string>>("Field_Names");
+  datafile=m_params.get<std::string>("Nudging_Filename");
+  time_step_file=m_params.get<Int>("Time_Step_File",-9999);
```

Comment:
Is this the timestep in the input NC file? Maybe we could read it straight from the NC file, rather than giving the impression to the user that they can change it? Perhaps, we could even read all the timesteps in the input file at init time, and store them in a std::vector, and compute the bef/aft time index based on the current time?

Reply:
Yes, this is the timestep in the input NC file. This is a good suggestion, I will implement this, thanks

Reply:
I initially implemented this std::vector, but after I implemented the struct with the before and after time_state data I actually didn't need it so I removed it. The NudgingData struct contains time information as well which is what I use to figure out if you need to load new data or not.